extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelper.java

```diff
@@ -669,40 +798,33 @@ public static boolean isHardcodedYoutubeMusicKeyValid() throws IOException,
-        final Map<String, List<String>> headers = new HashMap<>();
-        addCookieHeader(headers);
-        final String html = getDownloader().get(url, headers).responseBody();
-
-        String innertubeApiKey;
-        try {
-            innertubeApiKey = Parser.matchGroup1("INNERTUBE_API_KEY\":\"([0-9a-zA-Z_-]+?)\"", html);
-        } catch (final Parser.RegexException e) {
-            innertubeApiKey = Parser.matchGroup1("innertube_api_key\":\"([0-9a-zA-Z_-]+?)\"", html);
-        }
-
-        final String innertubeClientName
-                = Parser.matchGroup1("INNERTUBE_CONTEXT_CLIENT_NAME\":([0-9]+?),", html);
+        String musicClientVersion;
+        String musicKey;
+        String musicClientName;
 
-        String innertubeClientVersion;
         try {
-            innertubeClientVersion = Parser.matchGroup1(
-                    "INNERTUBE_CONTEXT_CLIENT_VERSION\":\"([0-9\\.]+?)\"", html);
-        } catch (final Parser.RegexException e) {
-            try {
-                innertubeClientVersion = Parser.matchGroup1(
-                        "INNERTUBE_CLIENT_VERSION\":\"([0-9\\.]+?)\"", html);
-            } catch (final Parser.RegexException ee) {
-                innertubeClientVersion = Parser.matchGroup1(
-                        "innertube_context_client_version\":\"([0-9\\.]+?)\"", html);
-            }
+            final String url = "https://music.youtube.com/sw.js";
+            final Map<String, List<String>> headers = new HashMap<>();
+            headers.put("Origin", Collections.singletonList("https://music.youtube.com"));
+            headers.put("Referer", Collections.singletonList("https://music.youtube.com"));
+            final String response = getDownloader().get(url, headers).responseBody();
+                musicClientVersion = getStringResultFromRegexArray(response,
+                        INNERTUBE_CONTEXT_CLIENT_VERSION_REGEXES, 1);
+                musicKey = getStringResultFromRegexArray(response, INNERTUBE_API_KEY_REGEXES, 1);
+                musicClientName = Parser.matchGroup1(INNERTUBE_CLIENT_NAME_REGEX, response);
+        } catch (final Exception e) {
+            final String url = "https://music.youtube.com/";
+            final Map<String, List<String>> headers = new HashMap<>();
+            addCookieHeader(headers);
+            final String html = getDownloader().get(url, headers).responseBody();
+
+            musicKey = getStringResultFromRegexArray(html, INNERTUBE_API_KEY_REGEXES, 1);
+            musicClientVersion = getStringResultFromRegexArray(html,
+                    INNERTUBE_CONTEXT_CLIENT_VERSION_REGEXES);
+            musicClientName = Parser.matchGroup1(INNERTUBE_CLIENT_NAME_REGEX, html);
         }
 
-        youtubeMusicKey = new String[]{
-                innertubeApiKey,
-                innertubeClientName,
-                innertubeClientVersion
-        };
+        youtubeMusicKey = new String[] {musicKey, musicClientName, musicClientVersion};
```

Comment:
Could we potentially introduce an object/class for this than using a String[] array? (To remove the need to guess what each index is)

Reply:
This should be done when we will refactor clients (with requests bodies buildings too, see https://github.com/TeamNewPipe/NewPipeExtractor/pull/780#discussion_r837881030).